source/Armament.cpp

```diff
@@ -51,7 +51,7 @@ void Armament::AddTurret(const Point &point, bool isUnder, const Outfit *outfit)
 int Armament::Add(const Outfit *outfit, int count)
 {
 	// Make sure this really is a weapon.
-	if(!count || !outfit || !outfit->IsWeapon())
+	if(!count || !outfit || !outfit->GetWeapon().IsWeapon())
```

Comment:
IsWeapon should probably be a function of Outfit.

Reply:
ok